process.go

```diff
@@ -0,0 +1,179 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"io"
+
+	"github.com/urfave/cli"
+)
+
+// == Direct input processing ========================================
+
+func processInputFiles(c *cli.Context, g *Gomplate) error {
+	inputs, err := readInputs(c.String("in"), c.StringSlice("file"))
+	if err != nil {
+		return err
+	}
+
+	outputs := c.StringSlice("out")
+	if len(outputs) == 0 {
+		outputs = []string{"-"}
+	}
+
+	for n, input := range inputs {
+		if err := renderTemplate(g, input, outputs[n]); err != nil {
```

Comment:
It probably makes sense cpu-wise, but I wonder whether io-wise it reasonable to write in parallel to disk. My gut feeling is that a serial approach might be not worse here 

But without a benchmark its hard to judge. 

Should I still change this ?  Also, this branch is only entered when using multiple --file options (not with the new --input-dir). So its not that many probably anyway.